src/proof_transcript.rs

```diff
@@ -130,12 +140,60 @@ impl ProofTranscript {
         Scalar::from_bytes_mod_order_wide(&buf)
     }
 
-    /// Pad separates the prior operations by padding
-    /// the rest of the block with zeroes and applying a permutation.
-    /// Each incoming message is length-prefixed anyway, but padding
-    /// enables pre-computing and re-using the oracle state.
-    fn pad(&mut self) {
+    /// Extracts a pair of challenge scalars.
+    ///
+    /// This is a convenience method that extracts 128 bytes and
+    /// reduces each 64-byte half modulo the group order.
+    pub fn challenge_scalars_pair(&mut self) -> (Scalar, Scalar) {
```

Comment:
Do we really need a custom function for this? It seems kind of non-orthogonal. If an API client wants to get `n` scalars and doesn't want to repeatedly call `challenge_scalar`, they could extract the bytes they want themselves.